backend/infrahub/core/diff/combiner.py

```diff
@@ -1,6 +1,319 @@
-from .model.path import EnrichedDiffRoot
+from copy import deepcopy
+from dataclasses import dataclass, field
+from typing import Iterable
+from uuid import uuid4
+
+from infrahub.core.constants import DiffAction, RelationshipCardinality
+from infrahub.core.schema_manager import SchemaManager
+
+from .model.path import (
+    EnrichedDiffAttribute,
+    EnrichedDiffNode,
+    EnrichedDiffProperty,
+    EnrichedDiffRelationship,
+    EnrichedDiffRoot,
+    EnrichedDiffSingleRelationship,
+)
+
+
+@dataclass
+class NodePair:
+    earlier: EnrichedDiffNode | None = field(default=None)
+    later: EnrichedDiffNode | None = field(default=None)
 
 
 class DiffCombiner:
-    def combine(self, earlier_diff: EnrichedDiffRoot, later_diff: EnrichedDiffRoot) -> EnrichedDiffRoot:  # pylint: disable=unused-argument
-        return earlier_diff
+    def __init__(self, schema_manager: SchemaManager) -> None:
+        self.schema_manager = schema_manager
+        # {child_uuid: (parent_uuid, parent_rel_name)}
```

Comment:
Should this be removed?

Reply:
I think this comment is helpful to understand the structure of this dict.
and it answers the question in your next comment